cmd/kubeadm/app/preflight/checks_linux.go

```diff
@@ -29,21 +29,19 @@ import (
 
 // Check validates if Docker is setup to use systemd as the cgroup driver.
 func (idsc IsDockerSystemdCheck) Check() (warnings, errorList []error) {
-	warnings = []error{}
 	driver, err := util.GetCgroupDriverDocker(exec.New())
 	if err != nil {
-		errorList = append(errorList, err)
-		return nil, errorList
+		return nil, append(errorList, err)
```

Comment:
what do you think of `[]error{err}`

Reply:
Thanks @draveness 
In fact, I did think about it. 
Using `append` can keep consistent with the code base, but `[]error{err}` maybe more efficient.
Simple Test:
```go
func M1() (errs []error) {
	return append(errs, fmt.Errorf("Err"))
}

func M2() (errs []error) {
	return []error{fmt.Errorf("Err")}
}

func Benchmark_M1(b *testing.B) {
	for i := 0; i < b.N; i++ {
		M1()
	}
}

func Benchmark_M2(b *testing.B) {
	for i := 0; i < b.N; i++ {
		M2()
	}
}
```
```sh
$ go test -test.bench=".*"
Benchmark_M1-4          10000000               210 ns/op
Benchmark_M2-4          10000000               199 ns/op
```
WDYT? @neolit123

Reply:
`append` will move the array to a new memory address. That's why it will be a little bit slower.

Reply:
`(warnings, errorList []error)` currently we have named variables here.

I'd suggest we only need to change to `([]error, []error)` and use `[]error{xxErr}`.

Reply:
> Using `append` can keep consistent with the code base, but `[]error{err}` maybe more efficient.

Please take a look at https://github.com/kubernetes/kubernetes/pull/81275/files#diff-a466e16a4b5d497feb0dc3539119ed25R173 which use `[]error{err}`

Reply:
> `(warnings, errorList []error)` currently we have named variables here.
> 
> I'd suggest we only need to change to `([]error, []error)` and use `[]error{xxErr}`.

have a named variable is useful for understanding, `([]error, []error)` could be confused.

Reply:
All right. I will change it later. Thanks @dixudx and @draveness